### ISSUE TITLE
Building clr shouldn't build host tests

### ISSUE THREAD

**Issue by eerhardt** [dotnet/runtime#73711]
Since https://github.com/dotnet/runtime/pull/73095, we are now building the `host` subset when building `clr`. However, the `host` subset includes a few test directories:

https://github.com/dotnet/runtime/blob/f244adbda9e680e39cfe6a660f695b07adec3792/eng/Subsets.props#L73

and

https://github.com/dotnet/runtime/blob/f244adbda9e680e39cfe6a660f695b07adec3792/src/native/corehost/CMakeLists.txt#L34

We should refactor our build so we don't need to build these tests when running a "normal" dev build: `.\build.cmd libs+clr -rc Release`.

**Comment by elinor-fung** [comment]
Should we make `clr` include `host.native` instead of `host`? If it is just being included to use the live host, I don't think we need host packages or tools either.

**Comment by am11** [comment]
Yes, `host.native` sounds like the right subset for clr and mono defaults. `Microsoft.NET.HostModel.dll` is loaded when `dotnet publish` is used with `--runtime` argument (with or without `-p:Publish{Aot,SingleFile,ReadyToRun}`). We only use live host's `dotnet publish` in the installer tests.

Ideally, if we are building all (or enough) subsets, then crossgen2 build (and anything else which uses `dotnet publish` from global sdk during the build) can be deferred after the $RuntimeFlavor+host+libs, and then `build.{cmd,sh}` can handover control from `runtime/.dotnet/dotnet` to live host in order to increase reliance on live build of corehost and HostModel.

**Comment by elinor-fung** [comment]
Switched to only include `host.native` in https://github.com/dotnet/runtime/pull/73800.

The native tests assets are part of `host.native` though, so they will still be built.
